drainer/syncer.go

```diff
@@ -466,12 +466,16 @@ ForLoop:
 				schema, table string
 			)
 			sql := b.job.Query
-			schema, table, err = s.schema.getSchemaTableAndDelete(b.job.BinlogInfo.SchemaVersion)
-			if err != nil {
-				err = errors.Trace(err)
-				break ForLoop
-			}
-			if s.cfg.SyncDDL {
+
+			if b.job.BinlogInfo.SchemaVersion == 0 {
+				log.Info("skip ddl due to the failed ddl", zap.String("sql", sql), zap.Int64("commit ts", commitTS))
+				shouldSkip = true
```

Comment:
I think just skipping this kind of jobs and then append fake binlogs is okay.
```suggestion
				appendFakeBinlogIfNeeded(nil, commitTS)
				continue
```